packages/app/src/heartbeatService.test.ts

```diff
@@ -273,6 +273,36 @@ describe('HeartbeatServiceImpl', () => {
       const emptyHeaders = await heartbeatService.getHeartbeatsHeader();
       expect(emptyHeaders).to.equal('');
     });
+    it('triggerHeartbeat() removes the earliest heartbeat once it exceeds the max number of heartbeats', async () => {
+      // Trigger heartbeats until we reach the limit
+      const numHeartbeats =
+        heartbeatService._heartbeatsCache?.heartbeats.length!;
+      for (let i = numHeartbeats; i <= 30; i++) {
+        await heartbeatService.triggerHeartbeat();
+        clock.tick(24 * 60 * 60 * 1000);
+      }
+
+      expect(heartbeatService._heartbeatsCache?.heartbeats.length).to.equal(30);
```

Comment:
Is it possible to make the max length a constant in the heartbeat service instead of using magic numbers?

Reply:
Done!